Cubical/Algebra/Group/Free.agda

```diff
@@ -0,0 +1,487 @@
+{-# OPTIONS --safe #-}
+
+module Cubical.Algebra.Group.Free where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Structure
+open import Cubical.Foundations.Function
+open import Cubical.Foundations.Equiv
+open import Cubical.Foundations.Isomorphism
+open import Cubical.Foundations.HLevels
+
+open import Cubical.Data.Bool as 𝟚 hiding (_≤_)
+open import Cubical.Data.Nat as ℕ hiding (_·_)
+open import Cubical.Data.Nat.Order.Recursive as OR
+open import Cubical.Data.Empty as ⊥
+open import Cubical.Data.Sum as ⊎
+open import Cubical.Data.Sigma
+open import Cubical.Data.List as Li
+
+open import Cubical.Relation.Nullary
+open import Cubical.Relation.Binary
+
+open import Cubical.Functions.Involution
+
+open import Cubical.Algebra.Group
+
+open import Cubical.HITs.SetQuotients as SQ renaming (_/_ to _/₂_ ; [_] to [_]/)
+
+
+
+private
+  variable
+    ℓ : Level
+
+[𝟚×_] : (A : Type ℓ) → Type ℓ
+[𝟚×_] = List ∘ (Bool ×_)
+
+module NormalForm (A : Type ℓ) where
+
+ -- Below we are defining predicate `HasRedex` over `List (Bool × A)`
+ -- basing it on presence of sublist - redex of form [(b , x) , (not b , x)]
```

Comment:
Maybe you should mention here, that "redex" means "reducible expression". (although it is implicit in what you write)

Reply:
modified comment to indicate that